src/unix/linux_like/mod.rs

```diff
@@ -24,6 +24,7 @@ s! {
         pub imr_interface: in_addr,
     }
 
+    #[cfg(not(any(target_arch = "mips64", target_arch = "sparc64")))]
     pub struct ip_mreqn {
```

Comment:
This won't work. If you don't want to add this to all linux targets, you need to manually add this to the targets you want. I do however recommend finding the source code for this and checking out what mips and sparc do.